src/main/java/net/fabricmc/meta/utils/ApiVersionParser.java

```diff
@@ -0,0 +1,48 @@
+package net.fabricmc.meta.utils;
+
+public class ApiVersionParser {
+
+    private String apiVersion;
+    private String minecraftVersion;
+
+    private String version;
+    private int build;
+
+    public ApiVersionParser(String version) {
+        this.version = version;
+
+        // Fix old API versions not having MC version
+        // Recreated from curseforge version index
+        if (version.contains("0.3.1") || version.contains("0.3.0")) {
+            version = version + "-1.14";
+        }
+
+        if (version.contains("+build.")) {
+            //TODO legacy remove when no longer needed
+            this.minecraftVersion = version.substring(version.lastIndexOf('-'));
+            this.apiVersion = version.substring(0, version.lastIndexOf('+') - 1);
+            this.build = Integer.parseInt(version.substring(version.lastIndexOf("+build.")+7, version.lastIndexOf('-')));
+        } else {
+            this.minecraftVersion = version.substring(version.lastIndexOf('+'));
+            this.apiVersion = version.substring(0, version.lastIndexOf('+') - 1);
+            this.build = 999; // Build number is no longer appended to versions, use dummy version
+        }
```

Comment:
This isnt going to be accrate and it needs to be if provided via an api. I've been over this a few times in detail in discord. 

The only way to provide good data is to have a manually managed database of what mc versions fabric api will work with. Some API versions work across a number of versions (Think pre-releases espeacially). this seems to just be using the branch provided in the version witch doesnt help decided between 1.16.1 and 1.16.2 in any way.

If launchers/installers are going to install fabric api they need to be 100% sure its going to work, this doesnt provide that.

Reply:
Where should this database exist, and how should it be added to?